nwb_conversion_tools/datainterfaces/icephys/abf/abfdatainterface.py

```diff
@@ -0,0 +1,142 @@
+"""Author: Luiz Tauffer."""
+from datetime import datetime, timedelta
+from pathlib import Path
+from warnings import warn
+import json
+from neo import AxonIO
+
+from ..baseicephysinterface import BaseIcephysInterface
+from ....tools.neo import get_number_of_electrodes, get_number_of_segments
+
+
+def get_start_datetime(neo_reader):
+    """Get start datetime for .abf file."""
+    if all(k in neo_reader._axon_info for k in ["uFileStartDate", "uFileStartTimeMS"]):
+        startDate = str(neo_reader._axon_info["uFileStartDate"])
+        startTime = round(neo_reader._axon_info["uFileStartTimeMS"] / 1000)
+        startDate = datetime.strptime(startDate, "%Y%m%d")
+        startTime = timedelta(seconds=startTime)
+        return startDate + startTime
+    else:
+        warn(
+            f"uFileStartDate or uFileStartTimeMS not found in {neo_reader.filename.split('/')[-1]}, datetime for "
+            "recordings might be wrongly stored."
+        )
+        return neo_reader._axon_info["rec_datetime"]
+
+
+class AbfInterface(BaseIcephysInterface):
+    """ABF IcephysInterface based on Neo AxonIO."""
+
+    neo_class = AxonIO
+
+    @classmethod
+    def get_source_schema(cls):
+        source_schema = super().get_source_schema()
+        source_schema["properties"]["file_paths"] = dict(
+            type="array",
+            minItems=1,
+            items={"type": "string", "format": "file"},
+            description="Array of paths to ABF files.",
+        )
+        source_schema["properties"]["metadata_file_path"] = dict(
+            type="string", format="file", description="Path to JSON file containing metadata for this experiment."
+        )
+        return source_schema
+
+    def __init__(self, file_paths: list, metadata_file_path: str = None):
+        # TODO: add docstring for init
+        super().__init__(file_paths=file_paths)
+        self.source_data["metadata_file_path"] = metadata_file_path
+
+    def get_metadata(self):
+        metadata = super().get_metadata()
+
+        metafile_data = dict()
+        metafile = self.source_data["metadata_file_path"]
+        if metafile is not None and Path(metafile).is_file():
+            with open(metafile) as json_file:
+                metafile_data = json.load(json_file)
+
+        # Extract start_time info
+        first_reader = self.readers_list[0]
+        first_session_time = get_start_datetime(neo_reader=first_reader)
+        session_start_time = first_session_time.strftime("%Y-%m-%dT%H:%M:%S%z")
+
+        # NWBFile metadata
+        if "NWBFile" not in metadata:
+            metadata["NWBFile"] = dict()
+        metadata["NWBFile"].update(
+            session_start_time=session_start_time, experimenter=[metafile_data.get("experimenter", "")]
```

Comment:
```suggestion
            session_start_time=session_start_time, experimenter=[metafile_data.get("experimenter", "")]
            if "experimenter" in metafile_data:
                metadata["NWBFile"].update(experimenter=metafile_data["experimenter"])
```

We shouldn't add a blank `experimenter` if it's not available.